src/parser/parser_support.c

```diff
@@ -8094,7 +8094,7 @@ pt_split_delete_stmt (PARSER_CONTEXT * parser, PT_NODE * delete_stmt)
     }
 
   /* if we have hints that refers globally to the join statement then we skip the split */
-  if ((delete_stmt->info.delete_.hint & PT_HINT_ORDERED && delete_stmt->info.delete_.ordered_hint == NULL)
+  if ((delete_stmt->info.delete_.hint & PT_HINT_ORDERED)
```

Comment:
I wonder if the results below are as intended.
```
drop table if exists t1, t2, t3, t4;

create table t1 (c1 int);
create table t2 (c1 int);
create table t3 (c1 int, c2 int);
create table t4 (c1 int, c2 int);
```
```
delete /*+ recompile leading(t1, t2) */ t1, t2, t4 from t1, t2, t3, t4 where t1.c1 = t2.c1 and t2.c1 = t3.c1;
  -> select /*+ LEADING(t1, t2) */ t2,class t2,t1,class t1 from t1 t1, t2 t2, t3 t3 where t1.c1=t2.c1 and t2.c1=t3.c1
  -> select /*+ LEADING */ t4,class t4 from t4 t4

delete /*+ recompile leading(t1, t2, t4) */ t1, t2, t4 from t1, t2, t3, t4 where t1.c1 = t2.c1 and t2.c1 = t3.c1;
  -> select t4,class t4,t2,class t2,t1,class t1 from t1 t1, t2 t2, t3 t3, t4 t4 where t1.c1=t2.c1 and t2.c1=t3.c1

delete /*+ recompile ordered */ t1, t2, t4 from t1, t2, t3, t4 where t1.c1 = t2.c1 and t2.c1 = t3.c1;
  -> select /*+ ORDERED */ t4,class t4,t2,class t2,t1,class t1 from t1 t1, t2 t2, t3 t3, t4 t4 where t1.c1=t2.c1 and t2.c1=t3.c1
```

Reply:
I think it would be better to ignore t4 mentioned in the hint in the second query and run it like the first query.

Reply:
I think it is correct to ignore the impossible join order. 
The t4 is classified into a different partition and a plan is created separately because it does not have a join term.
There seems to be no problem with adjusting join order.

Reply:
Since the hint was ignored, I think it is correct that a separate query should be executed for t4 in this case as well.
- Current PR
```
delete /*+ recompile leading(t1, t2, t4) */ t1, t2, t4 from t1, t2, t3, t4 where t1.c1 = t2.c1 and t2.c1 = t3.c1;
  -> [Warning: HINT ignored]
     select t4,class t4,t2,class t2,t1,class t1 from t1 t1, t2 t2, t3 t3, t4 t4 where t1.c1=t2.c1 and t2.c1=t3.c1
```
- My opinion
```
delete /*+ recompile leading(t1, t2, t4) */ t1, t2, t4 from t1, t2, t3, t4 where t1.c1 = t2.c1 and t2.c1 = t3.c1;
  -> [Warning: HINT ignored]
     select /*+ LEADING(t1, t2) */ t2,class t2,t1,class t1 from t1 t1, t2 t2, t3 t3 where t1.c1=t2.c1 and t2.c1=t3.c1
  -> [Warning: HINT ignored]
     select /*+ LEADING */ t4,class t4 from t4 t4
```